### DISCUSSION TITLE
[HW2] 2 Sparse Search

### DISCUSSION THREAD

**Discussion by emjay73** [seominjoon/kaist-ai605#100] [Q&A] [answered]
I'm sorry to interrupt the professor so many times but I'm quite new to NLP so I want to clarify the task.
So our retrieval task in the assignment is 
1. make vocab using SQUAD trainset.context only
2. convert validation set context into vector. (either using bow or tf-idf)
3. convert validation set question into vector.
4. find top k validation context index
5. among k candidates, only one index that is accompanied by a validation set question that I queried is the answer.

Is that correct?
And follow up questions..
1.  Does the ingredients for vocab generation includes title/answers/questions too?
2. when we compute idf for query, the number of docs =1? Or is it reasonable to compute "idf" using train set and reuse it for retrieval cuz this represents distribution?
4. how do I compute recall@k, when I do not know what's relevant?

**Comment by seominjoon** [comment, marked as answer]
Regarding the procedures:
Yes, what I intended is to use train set to make vocab and compute IDF for each term, and validation set for retrieval. You are right that only one context (document) will be the answer.

Regarding your questions:
1. This is a rather design choice, but it's fine not to include all of them.
2. You need to use train set to compute IDF and reuse it (without modifying it) during inference. Remember, computing IDF is done during training! Otherwise, it will be very slow.
3. The context that contains the question is the only relevant document. Recall@K simply measures if your top-k retrieval contains this context document.

**Reply by emjay73** [reply]
Things got more clear now. 
Thank you, professor!
And I left other posts with questions about the training set but it is not answered yet.
Please take a look at it.
https://github.com/seominjoon/kaist-ai605/discussions/98
https://github.com/seominjoon/kaist-ai605/discussions/109
Thank you.
